### PR TITLE
SFSettings | Add Agency Events menu item

### PR DESCRIPTION
**Pull request by Yopo-Nicolas**

### CODE REVIEW

File: src/Context/AgencyEvents.tsx
Comment (martinweingart) [inline src/Context/AgencyEvents.tsx:1]: There's no need for a context for now

File: src/Context/index.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export * from './Areas';
+export * from './AgencyEvents';
```
Comment (martinweingart) [inline src/Context/index.ts:2]: Remove

File: src/SFSettings.tsx [hunk 2]
```diff
@@ -290,6 +292,25 @@ export const SFSettings = ({
     ];
   }
 
+  if (!isRoleOfficer(user?.role.id) && useSubscription('shift')) {
```
Comment (martinweingart) [inline src/SFSettings.tsx:295]: Bad use of hook. Change useSubscription for getAppSubscription.

Check if the shifts subscription is used in another place and if so, assign it to a variable

File: src/Views/AgencyEvents/AgencyEvents.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+export interface AgencyEventsProps {
+  onClose: () => void;
+  onError: (e: SettingsError) => void;
+}
+
+export const AgencyEvents = ({
+  onClose,
+  onError
+}: AgencyEventsProps): React.ReactElement<AgencyEventsProps> => {
+  const { events } = React.useContext(AgencyEventsContext);
```
Comment (martinweingart) [inline src/Views/AgencyEvents/AgencyEvents.tsx:18]: Don't use context. Fech the events in a useEffect

File: src/Views/AgencyEvents/AgencyEvents.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  };
+
+  const getFilteredValues = (
+    list: AgencyEvent[],
+    filter: string
+  ): AgencyEvent[] => {
+    return list.filter((l) =>
+      l.name.toLowerCase().includes(filter.toLowerCase())
+    );
+  };
```
Comment (martinweingart) [inline src/Views/AgencyEvents/AgencyEvents.tsx:52]: This function can be defined outside the render function

File: src/Views/AgencyEvents/AgencyEvents.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    <div>
+      <SettingsContentRender
+        renderContent={() => (
+          <ListManagment
+            label="Event Type"
+            labelPlural="events"
+            list={events}
+            isLoading={isLoading}
+            filter={getFilteredValues}
+            onCreate={onFinish}
```
Comment (martinweingart) [inline src/Views/AgencyEvents/AgencyEvents.tsx:64]: `onCreate` should open the creation modal, not finishing a creation

File: src/Views/AgencyEvents/AgencyEventsList/AgencyEventsList.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import styles from './AgencyEventsList.module.scss';
+import { TransitionGroup } from 'react-transition-group';
+import { AgencyEvent } from '../../../Models';
+import { AgencyEventsListItem } from './AgencyEventsListItem/AgencyEventsListItem';
+import { SFCollapse } from 'sfui';
+import { Divider } from '../../../Components/Divider/Divider';
+
+export interface AgencyEventsListProps {
+  events: AgencyEvent[];
+  onView: (event: AgencyEvent) => void;
```
Comment (martinweingart) [inline src/Views/AgencyEvents/AgencyEventsList/AgencyEventsList.tsx:11]: Why this event?

File: src/Views/AgencyEvents/AgencyEvents.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    // TODO
+  };
+
+  const onDelete = (event: AgencyEvent) => {
+    // TODO
+  };
+
+  const onView = (event: AgencyEvent) => {
+    // TODO
+  };
```
Comment (martinweingart) [inline src/Views/AgencyEvents/AgencyEvents.tsx:31]: What it will do?

File: src/Views/AgencyEvents/AgencyEvents.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+            label="Event Type"
+            labelPlural="events"
+            list={events}
+            isLoading={isLoading}
+            filter={getFilteredValues}
+            onCreate={onFinish}
+            renderList={(list: AgencyEvent[]) => (
+              <AgencyEventsList
+                events={list}
+                onView={onView}
```
Comment (martinweingart) [inline src/Views/AgencyEvents/AgencyEvents.tsx:68]: Why this event?

File: src/Views/AgencyEvents/AgencyEventsList/AgencyEventsListItem/AgencyEventsListItem.tsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import styles from './AgencyEventsListItem.module.scss';
+import { SFIconButton, SFMenu, SFMenuItem, SFText } from 'sfui';
+import { AgencyEvent } from '../../../../Models';
+
+export interface AgencyEventsListItemProps {
+  event: AgencyEvent;
+  onClick: () => void;
```
Comment (martinweingart) [inline src/Views/AgencyEvents/AgencyEventsList/AgencyEventsListItem/AgencyEventsListItem.tsx:8]: Why this event?

**Review by martinweingart — CHANGES_REQUESTED** [review]

File: src/Context/AgencyEvents.tsx
Comment (Yopo-Nicolas) [reply src/Context/AgencyEvents.tsx:1]: Removed

File: src/Context/index.ts (hunk 1 above)
Comment (Yopo-Nicolas) [reply src/Context/index.ts]: Removed

File: src/SFSettings.tsx (hunk 2 above)
Comment (Yopo-Nicolas) [reply src/SFSettings.tsx]: Done, changed

File: src/Views/AgencyEvents/AgencyEvents.tsx (hunk 3 above)
Comment (Yopo-Nicolas) [reply src/Views/AgencyEvents/AgencyEvents.tsx]: Done

File: src/Views/AgencyEvents/AgencyEvents.tsx (hunk 4 above)
Comment (Yopo-Nicolas) [reply src/Views/AgencyEvents/AgencyEvents.tsx]: Moved

File: src/Views/AgencyEvents/AgencyEvents.tsx (hunk 5 above)
Comment (Yopo-Nicolas) [reply src/Views/AgencyEvents/AgencyEvents.tsx:63]: Changed

File: src/Views/AgencyEvents/AgencyEventsList/AgencyEventsList.tsx (hunk 6 above)
Comment (Yopo-Nicolas) [reply src/Views/AgencyEvents/AgencyEventsList/AgencyEventsList.tsx]: Removed

File: src/Views/AgencyEvents/AgencyEvents.tsx (hunk 7 above)
Comment (Yopo-Nicolas) [reply src/Views/AgencyEvents/AgencyEvents.tsx]: Removed

File: src/Views/AgencyEvents/AgencyEvents.tsx (hunk 8 above)
Comment (Yopo-Nicolas) [reply src/Views/AgencyEvents/AgencyEvents.tsx]: Removed

File: src/Views/AgencyEvents/AgencyEventsList/AgencyEventsListItem/AgencyEventsListItem.tsx (hunk 9 above)
Comment (Yopo-Nicolas) [reply src/Views/AgencyEvents/AgencyEventsList/AgencyEventsListItem/AgencyEventsListItem.tsx]: Removed

File: src/Views/AgencyEvents/AgencyEvents.tsx (hunk 5 above)
Comment (martinweingart) [reply src/Views/AgencyEvents/AgencyEvents.tsx:63]: I can't see the change

**Review by martinweingart — CHANGES_REQUESTED** [review]

File: src/Views/AgencyEvents/AgencyEvents.tsx (hunk 5 above)
Comment (Yopo-Nicolas) [reply src/Views/AgencyEvents/AgencyEvents.tsx]: Added missing change

**Review by martinweingart — APPROVED** [review]
LGTM

**Pull request merged**